src/coreservices.h

```diff
@@ -0,0 +1,140 @@
+#pragma once
+
+#include "preferences/configobject.h"
+#include "preferences/constants.h"
+#include "preferences/settingsmanager.h"
+#include "soundio/sounddeviceerror.h"
+#include "util/cmdlineargs.h"
+#include "util/timer.h"
+
+class QApplication;
+class CmdlineArgs;
+class KeyboardEventFilter;
+class EffectsManager;
+class EngineMaster;
+class SoundManager;
+class PlayerManager;
+class RecordingManager;
+class BroadcastManager;
+class ControllerManager;
+class VinylControlManager;
+class TrackCollectionManager;
+class Library;
+class LV2Backend;
+
+namespace mixxx {
+
+class DbConnectionPool;
+
+class CoreServices : public QObject {
+    Q_OBJECT
+
+  public:
+    CoreServices(const CmdlineArgs& args);
+    ~CoreServices() = default;
+
+    void initializeSettings();
+    // FIXME: should be private, but WMainMenuBar needs it initialized early
+    void initializeKeyboard();
+    void initialize(QApplication* pApp);
+    void shutdown();
+
+    std::shared_ptr<KeyboardEventFilter> getKeyboardEventFilter() const {
+        return m_pKeyboardEventFilter;
+    }
+
+    std::shared_ptr<ConfigObject<ConfigValueKbd>> getKeyboardConfig() const {
+        return m_pKbdConfig;
+    }
+
+    std::shared_ptr<SoundManager> getSoundManager() const {
+        return m_pSoundManager;
+    }
+
+    std::shared_ptr<PlayerManager> getPlayerManager() const {
+        return m_pPlayerManager;
+    }
+
+    std::shared_ptr<RecordingManager> getRecordingManager() const {
+        return m_pRecordingManager;
+    }
+
```

Comment:
I think it needs #ifdef __BROADCAST__
with -DBROADCAST=off:
In file included from /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/src/mixxx.h:8,
                 from /var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/src/mixxx.cpp:1:
/var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/src/coreservices.h: In member function ‘std::shared_ptr<BroadcastManager> mixxx::CoreServices::getBroadcastManager() const’:
/var/tmp/portage/media-sound/mixxx-9999/work/mixxx-9999/src/coreservices.h:64:16: error: ‘m_pBroadcastManager’ was not declared in this scope; did you mean ‘getBroadcastManager’?
   64 |         return m_pBroadcastManager;
      |                ^~~~~~~~~~~~~~~~~~~
      |                getBroadcastManager

Reply:
Yes. Right, I will take care.

Reply:
https://github.com/mixxxdj/mixxx/pull/3514